tests/0102-static_group_rebalance.c

```diff
@@ -0,0 +1,323 @@
+        rd_kafka_t *rk;
+        test_msgver_t *mv;
+        int64_t assigned_at;
+        int64_t revoked_at;
+        int partition_cnt;
+        rd_kafka_resp_err_t expected_rb_event;
+} _consumer_t;
+
+
+/**
+ * @brief Call poll until a rebalance has been triggered
+ */
+int static_member_wait_rebalance (_consumer_t *c, int64_t start,
+                                  int64_t *target, int timeout_ms) {
+        int64_t tmout = test_clock() + (timeout_ms * 1000);
+
+        while (timeout_ms < 0 ? 1 : test_clock() <= tmout) {
+                if (*target > start)
+                        return 1;
+                test_consumer_poll_once(c->rk, c->mv, 1000);
+        }
+
+        return 0;
+}
+
+static void rebalance_cb (rd_kafka_t *rk,
+                          rd_kafka_resp_err_t err,
+                          rd_kafka_topic_partition_list_t *parts,
+                          void *opaque) {
+        int i;
+        _consumer_t *c = opaque;
+
+        TEST_ASSERT(c->expected_rb_event == err,
+                    "%s: Expected rebalance event %s got %s\n",
+                    rd_kafka_name(rk),
+                    rd_kafka_err2name(c->expected_rb_event),
+                    rd_kafka_err2name(err));
+
+        switch (err)
+        {
+        case RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS:
+                TEST_SAY("%s Assignment (%d partition(s)): ",
+                         rd_kafka_name(rk), parts->cnt);
+                for (i = 0 ; i < parts->cnt ; i++)
+                        TEST_SAY0("%s%s[%"PRId32"]",
+                                  i == 0 ? "" : ", ",
+                                  parts->elems[i].topic,
+                                  parts->elems[i].partition);
+                TEST_SAY0("\n");
+
+                c->partition_cnt = parts->cnt;
+                c->assigned_at = test_clock();
+                rd_kafka_assign(rk, parts);
+
+                break;
+
+        case RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS:
+                c->revoked_at = test_clock();
+                rd_kafka_assign(rk, NULL);
+                TEST_SAY("%s revoked %d partitions\n", 
+                         rd_kafka_name(c->rk), parts->cnt);
+
+                break;
+
+        default:
+                TEST_FAIL("rebalance failed: %s", rd_kafka_err2str(err));
+                break;
+        }
+
+        /* prevent poll from triggering more than one rebalance event */
+        rd_kafka_yield(rk);
+}
+
+int main_0102_static_group_rebalance (int argc, char **argv) {
+        rd_kafka_conf_t *conf;
+        test_msgver_t mv;
+        int64_t rebalance_start;
+        _consumer_t c[_CONSUMER_CNT] = RD_ZERO_INIT;
+        const int msgcnt = 100;
+        uint64_t testid  = test_id_generate();
+        const char *topic = test_mk_topic_name("0102_static_group_rebalance",
+                                               1);
+        char *topics = rd_strdup(tsprintf("^%s.*", topic));
+
+        test_conf_init(&conf, NULL, 60);
```

Comment:
What's the runtime of the test today? Is it close to 60?
It is usually a good idea to allow an extra 20-30s for long tests like this to make it more robust.

Reply:
Typically 34-40 seconds locally. Which would be 20-30s less than 60. I can check the travis output though to ensure it aligns with that

Reply:
Bumped to 70 since the additional session timeout test more consistently takes 40 seconds